### ISSUE TITLE
Consider extracting the currently visible attribute in the CUDS powered sources   

### ISSUE THREAD

**Issue by itziakos** [simphony/simphony-mayavi#35]
The aim is to reduce the memory footprint during visualization.

**Comment by itziakos** [comment]
To support such a behavior I can think of two design solutions
1. A `C++` (probably `Cython`) wrapper of CUDS containers to vtkDataSet.
2. A Special Source that copies the structure of a CUDS container and then extracts the active DataArray dynamically.

(1) will provide the biggest memory savings while (2) might be easier to prototype.

**Comment by itziakos** [comment]
Move this out of the Milestone 0.2.0 since it will require some investigation

**Comment by stefanoborini** [comment]
I investigated this and I don't think it's trivial. There are two issues:
1. the assumption (and this is a design choice documented in the wiki) is that data are always copied, and there should be no reference to the originating source. For example, the VTKParticle object receives data copied from the originating cuds at construction. All data are copied and the resulting vtk cuds is installed.
2. Once the creation is done, the VTK cuds object has an interface that allows modification (e.g. remove or add particles).

If I install only the data for, say, TEMPERATURE, remove a particle from the vtk cuds, then install data for RADIUS, there are many points of potential attrition:
1. the cuds must be the same we used to initialize the vtk cuds.
2. the vtk cuds now contains one particle less than the source cuds, so the installation of the RADIUS requires bookkeeping. I am not sure we have the data to allow for this bookkeeping.
3. the alternative would be to create a new vtk cuds every time, for a single cuds data, but this would discard added/removed particles every time we change the data we want to visualise.

I am unsure how heavy would be to introduce these changes, but at the stage of uncertainty of the rest of the code, I need a bit more exploration to decide how to remove the assumption without breaking the rest.

Plus, I am not even sure we actually use the vtk cuds interface that adds/removes things.... a quick investigation says we don't, but I might be wrong, which means that there's no reason to allow that class to be mutable.

**Comment by stefanoborini** [comment]
I have to give up on this. The whole design is made so that the constraint of having that data is verified. For example, installing partial data, and then having get_particle() called would return a partially filled particle. 

The only way for making this truly opaque is to let the VTK cuds object keep a reference to the original CUDS, so that it can preemptively plug in stuff on demand when requested. The assumption is that we never do that.

Summing up, I would not do this issue, because it's a direct and inevitable consequence on how we decided to handle data transit in the application.

**Comment by stefanoborini** [comment]
After discussion with @itziakos many stopping conditions are gone. I am revisiting this after a few preparing PRs
